interfaces/saml/v0/interface_tests/test_provider.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+from interface_tester.interface_test import Tester
+from scenario import Relation, State
+
+
+def test_data_published_on_created():
+    t = Tester(State(
+        leader=True,
```

Comment:
leader=True is mandated charm-side, not interface-side. 
The interface doesn't care if the remote end is leader, only that it publishes the right data.

Reply:
so you can remove it from here, it will be the charm's duty to put it in its interface_tester's state_template.